### ISSUE TITLE
segfault using nix run -i

### ISSUE THREAD

**Issue by mmerickel** [NixOS/nix#1907]
For some reason `--ignore-environment` causes `nix run` to segfault on my system.

```
~❯ nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy
querying info about missing paths[1]    47046 segmentation fault  nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy
~❯ nix run -f '<nixpkgs>' python3 python36.pkgs.numpy
bash-3.2$
```

```
~❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 17.4.0, macOS 10.13.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(michael): `""`
 - channels(root): `"nixpkgs-18.03pre129189.c2fbd472a4e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by dtzWill** [comment]
For what it's worth, the given command doesn't work on Linux either because it tries to run 'bash':

```console
$ nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy
error: unable to exec 'bash': No such file or directory
```

Which could probably be cleaned up a bit, not sure.

Does specifying the command to run prevent the segfault?

```console
$ nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy -c python
```
?

**Comment by edolstra** [comment]
@dtzWill You need to add `bashInteractive` to the set of packages. Also, expect warnings due to $HOME not being set.

**Comment by mmerickel** [comment]
Specifying the command does not help.

```
~❯ nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy -c python
querying info about missing paths[1]    57584 segmentation fault  nix run -i -f '<nixpkgs>' python3 python36.pkgs.numpy -c python
```

**Comment by jcrben** [comment]
@edolstra I see that the commit which closes this says `Note that clearenv() is not available on macOS.` - I'm guessing that means the `-i` flag is not expected to work in macOS? is that documented outside of the commit message?

**Comment by dtzWill** [comment]
No, it explains why we implement `clearEnv` instead of calling `clearenv()`-- see [`man 3 clearenv`](https://linux.die.net/man/3/clearenv).

Are you observing the `-i` flag not working on macOS?

**Comment by jcrben** [comment]
👍 

I got some patience and waited for `fetching Git...` to end (had originally been wondering about progress and stopped too soon). From there, I built per https://nixos.org/nix/manual/#chap-hacking and the fix seems to be working. 
![image](https://user-images.githubusercontent.com/5614134/37572032-6f9cee88-2ac2-11e8-8644-97980381aa1c.png)
![image](https://user-images.githubusercontent.com/5614134/37572034-7695ee2e-2ac2-11e8-8417-ae5146088f15.png)

There's a different error, but I think that's probably more expected / user error on my part. Still a bit lost as whole.

I figure the error in the released 2.0 install is to be expected - I did the recommended curl install script today (https://nixos.org/nix/download.html).
